src/object/srv_ec_aggregate.c

```diff
@@ -2235,6 +2236,8 @@ agg_object(daos_handle_t ih, vos_iter_entry_t *entry,
 		if (rc < 0) {
 			D_ERROR("oid:"DF_UOID" ds_pool_check_leader failed "
 				DF_RC"\n", DP_UOID(entry->ie_oid), DP_RC(rc));
+			if (rc == -DER_STALE)
+				agg_param->ap_obj_skipped = 1;
```

Comment:
Since skipping object can't be tolerated, why don't we just simply return error here?

Reply:
If some object cannot be rebuilt/recovered, for example EC 4+2 lost 3 shards, for these cases we can skip those obj's aggregation, rather than block all objects. right?